src/main/java/com/limechain/utils/scale/ScaleUtils.java

```diff
@@ -0,0 +1,71 @@
+package com.limechain.utils.scale;
+
+import com.limechain.utils.scale.writers.PairWriter;
+import io.emeraldpay.polkaj.scale.ScaleCodecReader;
+import io.emeraldpay.polkaj.scale.ScaleCodecWriter;
+import io.emeraldpay.polkaj.scale.ScaleReader;
+import io.emeraldpay.polkaj.scale.reader.ListReader;
+import io.emeraldpay.polkaj.scale.writer.ListWriter;
+import kotlin.Pair;
+import lombok.experimental.UtilityClass;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
+
+// TODO: This class is currently just a helper utility class,
+//  planned to grow into a unified scale encode/decode util class with whatever methods are useful
+//  WIP
+// Currently trying out different approaches to spare some of the boilerplate around SCALE en/decoding
```

Comment:
Indentation is a big weird, why not have it all start at the left. You don't need a comma after class and you delete the first "class" and the "just" strings.